pkg/kubelet/cm/memorymanager/policy_static.go

```diff
@@ -45,6 +46,10 @@ type staticPolicy struct {
 	systemReserved systemReservedMemory
 	// topology manager reference to get container Topology affinity
 	affinity topologymanager.Store
+	// reusableState contains the state before applying init containers allocations
+	reusableState map[string]state.NUMANodeMap
+	// initContainersReusableMemory contains the memory allocated for init containers that can be reused
+	initContainersReusableMemory reusableMemory
```

Comment:
Maybe it's just the naming, but I'm confused as to the purpose of `reusableState`. Can you elaborate on what it is and why it's necessary in addition to `initContainersReusableMemory`?

Reply:
**reusableState**
Because init containers start once the previous init containers finished to run and the same true for the first app container I find it much easier to save the state during the first pod container allocation under `reusableState` and pass this state to GetTopologyHints, so Allocation for init containers will not affect hints calculation for others init containers and the first app container under the same pod, than updating the `calculateHints` method to be aware of re-usable memory.

**initContainersReusableMemory**
I still need to track the init containers re-usable memory because, in the end, I will need to reflect correctly this amount of allocated memory under the state file.

Please give me know if you need more information.